lua/main_config.lua

```diff
@@ -23,14 +23,18 @@ function gatekeeper_init()
 
 	local numa_table = staticlib.get_numa_table(net_conf)
 
+	local n_fixed_lcores = gatekeeper_server and 5 or 3
+	local fixed_lcores = staticlib.convert_table_to_array(
+		staticlib.alloc_lcores_evenly_from_all_numa_nodes(numa_table,
+			n_fixed_lcores, 0))
```

Comment:
This whole patch can be simplified with the following steps:
1. Rename `fixed_lcores` to `aux_numa_table` and keep the return of `alloc_lcores_evenly_from_all_numa_nodes()` as is, that is, we don't need `convert_table_to_array()` here;
2. For every occasion we need a lcore, use `staticlib.alloc_an_lcore(aux_numa_table)` instead of `table.remove(fixed_lcores)`.

This solution avoids editing files `lua/lls.lua`, `lua/cps.lua`, and `lua/dyn_cfg.lua`, avoids dropping `alloc_an_lcore()`  (i.e. no editing to file `lua/gatekeeper/staticlib.lua`).